src/pkg/cluster/injector.go

```diff
@@ -53,15 +56,15 @@ func (c *Cluster) StartInjectionMadness(tmpDir string, imagesDir string, injecto
 	}
 
 	var err error
-	var images k8s.ImageNodeMap
+	var images imageNodeMap
 	var payloadConfigmaps []string
 	var sha256sum string
 	var seedImages []transform.Image
 
 	// Get all the images from the cluster
 	timeout := 5 * time.Minute
 	spinner.Updatef("Getting the list of existing cluster images (%s timeout)", timeout.String())
-	if images, err = c.GetAllImages(timeout, injectorRequestedCPU, injectorRequestedMemory); err != nil {
+	if images, err = c.GetImagesFromAvailableNodes(timeout); err != nil {
```

Comment:
```suggestion
	if images, err = c.getImagesAndNodesForInjection(timeout); err != nil {
```
This also has to change